presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,4 +728,277 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            return isMaterializedView(table);
+        }
+        return false;
+    }
+
+    @Override
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        Optional<Table> materializedViewOptional = metastore.getTable(new HiveIdentity(session), viewName.getSchemaName(), viewName.getTableName());
+        if (!materializedViewOptional.isPresent()) {
+            return Optional.empty();
+        }
+        if (!isMaterializedView(session, viewName)) {
+            return Optional.empty();
+        }
+
+        Table materializedView = materializedViewOptional.get();
+
+        ConnectorMaterializedViewDefinition definition = decodeMaterializedViewData(materializedView.getViewOriginalText()
+                .orElseThrow(() -> new PrestoException(HIVE_INVALID_METADATA, "No view original text: " + viewName)));
+
+        String storageTable = materializedView.getParameters().getOrDefault(STORAGE_TABLE, "");
+        return Optional.of(new ConnectorMaterializedViewDefinition(
+            definition.getOriginalSql(),
+            storageTable,
+            definition.getCatalog(),
+            Optional.of(viewName.getSchemaName()),
+            definition.getColumns(),
+            definition.getComment(),
+            Optional.of(materializedView.getOwner()),
+            new HashMap<>(materializedView.getParameters())));
+    }
+
+    public Optional<TableToken> getTableToken(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        return Optional.ofNullable(icebergTable.currentSnapshot())
+            .map(snapshot -> new TableToken(snapshot.snapshotId()));
+    }
+
+    public boolean isTableCurrent(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<TableToken> tableToken)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        Optional<TableToken> currentToken = getTableToken(session, handle);
+
+        if (!tableToken.isPresent() || !currentToken.isPresent()) {
+            return false;
+        }
+
+        return tableToken.get().getSnapshotId() == currentToken.get().getSnapshotId();
+    }
+
+    @Override
+    public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession session, SchemaTableName materializedViewName)
+    {
+        Map<String, Optional<TableToken>> refreshStateMap = getMaterializedViewToken(session, materializedViewName);
+        if (refreshStateMap.isEmpty()) {
+            return new MaterializedViewFreshness(false);
+        }
+
+        for (Map.Entry<String, Optional<TableToken>> entry : refreshStateMap.entrySet()) {
+            List<String> strings = Splitter.on(".").splitToList(entry.getKey());
+            checkState(strings.size() >= 2);
```

Comment:
We should be more explicit in the validation here and provide a good error message, since this value comes from the user. Let's move this into `getMaterializedViewToken()` and have it return `SchemaTableName` since that is where we are doing the parsing.
```java
if (strings.size() == 3) {
    ...
}
else if (strings.size() != 2) {
    throw new PrestoException(ICEBERG_INVALID_METADATA, format("Invalid table name in '%s' property: %s'", DEPENDS_ON_TABLES, name));
}

Reply:
Added the validation. Leaving the code where it is as we expect `GetMaterializedViewFreshness` to evolve over time.